### ISSUE TITLE
Can't set cluster private registry password in UI

### ISSUE THREAD

**Issue by gaktive** [rancher/rancher#38819]
Internal reference: SURE-4978
Reported in 2.6.4 & 2.6.6

Issue description:
If you edit a downstream cluster from the UI to add a private registry, the password provided does not apply. The same problem happens if you Edit Config > Edit as YAML. The UI or editing as YAML should allow you to enter a password and have it work.

Business impact:
After upgrading, the user's images from private registries were not able to be pulled causing errors in the workloads of their downstream clusters.

Repro steps:

1. Create a custom downstream cluster
2. From cluster management edit config for the cluster
3. In the Private Registry section, check enabled and add docker.io as the default
4. Cluster will update. Then Edit as YAML
5. Add another private registry so it looks like this:

```
  private_registries:
    - ecr_credential_plugin: {}
      is_default: true
      url: docker.io
    - ecr_credential_plugin: {}
      is_default: false
      url: sample.domain.name.tld
      user: user@domain.name.tld
      password: <PASSWORD>
```

6. Save and the cluster doesn't seem to update
7. Try to create a workload that comes from the private registry added in step 5 -- the image won't pull
8. Go back to the cluster config and try to enter the password in the "Private Registry" section -- it won't apply and workloads still can't pull images from the private registry
9. Edit the cluster with `kubectl edit clusters.management.cattle.io c-XXXXX` -- the cluster updates and then you can pull images from the private registry
10. Interestingly, if you now change the password in the UI to a wrong password the cluster updates and the images stop pulling again. I'm not sure why this would cause an update but step 5 and 8 did not

Workaround:
Edit the CRD for the cluster, see step 9 above

Actual behavior:
The cluster private registry password is not applied from GUI / YAML

Expected behavior:
The cluster private registry password should be applied GUI / YAML

Additional notes:
This worked in 2.5.11

**Comment by HarrisonWAffel** [comment]
sync'd up with gaktive and trasnferred this ticket to rancher/rancher, as it does not seem to be a UI issue. 

> I asked Seb about it - it sounds like the breaking RKE1 change probably happened here: #36578

I would agree with this, this seems like a regression introduced when we changed how private registry secrets were handled. I found a similar issue regarding multiple private registries for RKE1 https://github.com/rancher/rancher/issues/37658, so this use case is supported but has had some trouble in the past due to how the secrets are created/handled by rancher.

**Comment by rmweir** [comment]
Cannot consistently reproduce. I believe there is some sort of race condition with passing the docker credentials to RKE and migrating the registry secrets.

Before migration I believe the credentials were passed through the rke object to rke provisioner and this contained the credentials. If this part of the code receives the rke object before the registry password is replaced with a secret then workloads will be able to access the repo. If migration has happened first then the way the credentials are passed is through redeploying the agent which contains manifest code for a docker credential secret. I think the redeploy code is missing logic that considers registries being updated that aren't the first in the slice to trigger a redeploy: https://github.com/rancher/rancher/blob/release/v2.7/pkg/controllers/management/clusterdeploy/clusterdeploy.go#L300.

It also looks like the docker credential secret that is created only contains info for the first registry in the slice of registries: https://github.com/rancher/rancher/blob/release/v2.7/pkg/systemtemplate/import.go#L92.

**Comment by cbron** [comment]
@HarrisonWAffel @Sahota1225 passing this back to team 2, as it is trending towards wider changes around how rke1 handles multiple registries. Happy to meet up and chat about it.

**Comment by snasovich** [comment]
This has been resolved sometime before v2.7.8 as it's no longer reproducible on that (or latest) versions. Most likely it was fixed in v2.7.1 but we didn't go that far to test.
